### PR TITLE
Cli very basic setup with first command

### PR DESCRIPTION
**Pull request by steffnay**
Here is what I have so far for the retrieveIssues command. 
Please let me know what sort of changes I should make as I'm writing the others.

I'm also a little hung up on the csv that is created with Json2csvParser. I tried a couple others, but they also made very ugly files. Wondering if there is a better way to clean up the text that comes from the issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1 @@
-*.sublime-*
-node_modules/
```
Comment (JustinBeckwith) [inline .gitignore:2]: It's a good idea to keep `node_modules` in your gitignore :)

File: cli.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+/**
+ * Takes a filepath to a json object of repositories
+ * and a filename to save the resulting issue data,
+ * then makes api call to GitHub to retrieve current issues
+ * @param {string} data
+ * @param {string} file
+ */
+async function retrieveIssues(data, file) {
+  let rawdata = fs.readFileSync(data,'utf8')
```
Comment (JustinBeckwith) [inline cli.js]: You can use const here and below, right?

File: cli.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    issueInfo.labels = labelNames;
+
+    return issueInfo;
+  }
+}
+
+// extracts the value for the name key from each label object
+// and returns an array of all the issue's labels
+function separateLabels(labelsArray){
+  let newArray = []
```
Comment (JustinBeckwith) [inline cli.js]: You should use a `map` here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

**Review by JustinBeckwith — COMMENTED** [review]
Nice first PR here 🎉 I can take another look once `node_modules` is added to .gitignore, and deleted from the PR

File: .gitignore (hunk 1 above)
Comment (steffnay) [reply .gitignore:2]: ah thanks!

Comment (steffnay) [reply .gitignore:2]: Okay, this should be fixed.

File: cli.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const axios = require('axios');
+const async = require("async");
```
Comment (JustinBeckwith) [inline cli.js]: I don't think you actually need the `async` library.

File: package.json [hunk 5]
```diff
@@ -8,53 +8,31 @@
...
-  "devDependencies": {
-    "chai": "3.3.0",
-    "coveralls": "^2.11.4",
-    "faker": "3.0.1",
-    "istanbul-harmony": "0.3.16",
-    "keygrip": "1.0.1",
-    "mocha": "2.3.3",
-    "sinon": "1.17.1",
-    "supertest": "1.1.0"
+    "async": "^2.6.1",
```
Comment (JustinBeckwith) [inline package.json]: For most modern nodejs, you likely don't need to use the `async` npm module.

File: cli.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  let inssueInfo = {}
+
+  const githubClientID = settings.githubClientID;
+  const githubClientSecret = settings.githubClientSecret;
+
+  let url = `https://api.github.com/repos/${repoName}/issues/${number}?client_id=${githubClientID}&client_secret=${githubClientSecret}`;
+
+  let response = await axios.get(url);
+  // should this have different response to error?
+  if (response.err) { console.log('error'); }
```
Comment (JustinBeckwith) [inline cli.js]: When an async function throws, like what you're doing above with `await axios.get(url)`, it's typically going to throw an exception which exits the function, and crashes the main process.  If you want to gracefully handle this, you would have to wrap the `await` line with a `try/catch`.  Let me know if this doesn't make sense :)

Comment (steffnay) [reply cli.js]: I think I got it with this one. Let me know if I misunderstood. I also should have addressed all of the other issues you mentioned.

File: .gitignore [hunk 7]
```diff
@@ -1,6 +1,2 @@
...
-key.json
-npm-debug.log
-config.js
-coverage/
\ No newline at end of file
+settings.json
```
Comment (jmdobry) [inline .gitignore:1]: It's probably best if you just leave this file as-is. In fact, keep the original content of this file and add `.DS_Store` and `settings.json` to it

File: cli.js [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+      // const projectID = settings.projectID;
+      // const computeRegion = settings.computeRegion;
+      // const datasetName = opts.datasetName;
+      // const multiLabel = opts.multilabel;
+      //
+      // createDataset(projectID, computeRegion, datasetName, multiLabel)
+    }
+  )
+  .command(
+    `importData <issuesDataFilePath> <datasetID>`,
```
Comment (jmdobry) [inline cli.js:37]: nit: s/`datasetID`/`datasetId`/

File: cli.js [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+    `Import the GitHub issues data into the Google AutoML NL dataset by specifying the file of issues  data and the dataset ID.`,
+    {},
+    opts => {
+//       const projectID = settings.projectID;
+//       const computeRegion = settings.computeRegion;
+//       const datasetId = opts.datasetID;
+//       const path = opts.issuesDataFilePath;
+// //  TODO: SHOULD THIS READ FROM LOCAL FILE OR REQUIRE GCS UPLOAD LIKE CODE SAMPLE?
+//       importData(projectID, computeRegion, datasetId, path);
```
Comment (jmdobry) [inline cli.js:46]: nit: s/`projectID`/`projectId`/

File: cli.js [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+    }
+  )
+  .command(
+    `createModel <datasetID> <modelName> <GCPprojectID> <GCPcomputeRegion>`,
```
Comment (jmdobry) [inline cli.js:50]: nit: s/`datasetID`/`datasetId`/

nit: s/`GCPprojectID`/`projectId`/

nit: s/`GCPcomputeRegion`/`computeRegion`/

File: cli.js [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+    `Train the Google AutoML NL model by specifying the dataset ID generated, the name for new model being generated, the ID of the Google Cloud Platform project, and the region from Google Cloud Platform project.`,
+    {},
+    opts => {
+        //TODO: create function
```
Comment (jmdobry) [inline cli.js:54]: Nit: Add `console.log('Not yet implemented!');`

File: cli.js [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+    }
+  )
+  .command(
+    `analyzeAccuracy <issuesDataFilePathilepath> <repo>`,
+    `Calculate the accuracy of the model's predictons by comparing LabelCat labels applied upon issue opening with the labels present when issue was closed.`,
+    {},
+    opts => {
+        //TODO: create function
```
Comment (jmdobry) [inline cli.js:62]: Nit: Add `console.log('Not yet implemented!');`

File: cli.js (hunk 9 above)
Comment (jmdobry) [reply cli.js:46]: Nit: Add `console.log('Not yet implemented!');`

File: cli.js [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+    `createDataset <datasetName> [multilabel]`,
+    `Create a new Google AutoML NL dataset with the specified name.`,
+    {},
+    opts => {
+      // const projectID = settings.projectID;
+      // const computeRegion = settings.computeRegion;
+      // const datasetName = opts.datasetName;
+      // const multiLabel = opts.multilabel;
+      //
+      // createDataset(projectID, computeRegion, datasetName, multiLabel)
```
Comment (jmdobry) [inline cli.js:33]: Nit: Add `console.log('Not yet implemented!');`

File: cli.js [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+    }
+  )
+  .example(`$0 retrieveIssues repoData.json issuesData.csv`, `Retrieves issues from list of repos in repoData.json and saves the resulting information to issuesData.csv.`)
+  .example(`$0 createDataset Data`, `Creates a new dataset with the specified name.`)
+  .example(`$0 importData issuesData.csv 1248102981`, `Imports the GitHub issues data into the dataset by specifying the file of issues data and the dataset ID.`)
+  .example(`$0 createModel 22093823 “test model” 23423423 us-central1`, `Trains the Google AutoML NL model by specifying the dataset ID generated, the name for new model being generated, the ID of the Google Cloud Platform project, and the region from Google Cloud Platform project.`)
+  .example(`$0 analyzeAccuracy issuesData.csv some-repo`, `Calculates the accuracy of the model's predictons by comparing LabelCat labels applied upon issue opening with the labels present when issue was closed.`)
+  .wrap(120)
+  .recommendCommands()
+  .epilogue(`For more information, contact....`)
```
Comment (jmdobry) [inline cli.js:72]: Delete

File: cli.js [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+    {},
+    opts => {
+        //TODO: create function
+    }
+  )
+  .example(`$0 retrieveIssues repoData.json issuesData.csv`, `Retrieves issues from list of repos in repoData.json and saves the resulting information to issuesData.csv.`)
+  .example(`$0 createDataset Data`, `Creates a new dataset with the specified name.`)
+  .example(`$0 importData issuesData.csv 1248102981`, `Imports the GitHub issues data into the dataset by specifying the file of issues data and the dataset ID.`)
+  .example(`$0 createModel 22093823 “test model” 23423423 us-central1`, `Trains the Google AutoML NL model by specifying the dataset ID generated, the name for new model being generated, the ID of the Google Cloud Platform project, and the region from Google Cloud Platform project.`)
+  .example(`$0 analyzeAccuracy issuesData.csv some-repo`, `Calculates the accuracy of the model's predictons by comparing LabelCat labels applied upon issue opening with the labels present when issue was closed.`)
```
Comment (jmdobry) [inline cli.js:69]: Change `some-repo` to look more like an actual repo name, like `SomeOrg/SomeRepo`

File: cli.js [hunk 16]
```diff
@@ -0,0 +1,263 @@
...
+    `Calculate the accuracy of the model's predictons by comparing LabelCat labels applied upon issue opening with the labels present when issue was closed.`,
+    {},
+    opts => {
+        //TODO: create function
+    }
+  )
+  .example(`$0 retrieveIssues repoData.json issuesData.csv`, `Retrieves issues from list of repos in repoData.json and saves the resulting information to issuesData.csv.`)
+  .example(`$0 createDataset Data`, `Creates a new dataset with the specified name.`)
+  .example(`$0 importData issuesData.csv 1248102981`, `Imports the GitHub issues data into the dataset by specifying the file of issues data and the dataset ID.`)
+  .example(`$0 createModel 22093823 “test model” 23423423 us-central1`, `Trains the Google AutoML NL model by specifying the dataset ID generated, the name for new model being generated, the ID of the Google Cloud Platform project, and the region from Google Cloud Platform project.`)
```
Comment (jmdobry) [inline cli.js:68]: Change your example `projectId` to look more like a real project ID, such as `my-project-1234`

File: cli.js [hunk 17]
```diff
@@ -0,0 +1,263 @@
...
+/**
+ * Takes a filepath to a json object of repositories
+ * and a filename to save the resulting issue data,
+ * then makes api call to GitHub to retrieve current issues
+ * @param {string} data
+ * @param {string} file
+ */
+async function retrieveIssues(data, file) {
+  const rawdata = fs.readFileSync(data,'utf8')
+  let issues = JSON.parse(rawdata);
```
Comment (jmdobry) [inline cli.js:87]: `issues` isn't re-assigned, use `const` instead

File: cli.js [hunk 18]
```diff
@@ -0,0 +1,263 @@
...
+ * @param {object} issue
+ */
+async function getIssueInfo(issue) {
+  const repoName = issue.Repo;
+  const number = issue.IssueID;
+
+  const githubClientID = settings.githubClientID;
+  const githubClientSecret = settings.githubClientSecret;
+
+  let url = `https://api.github.com/repos/${repoName}/issues/${number}?client_id=${githubClientID}&client_secret=${githubClientSecret}`;
```
Comment (jmdobry) [inline cli.js:112]: `url` isn't re-assigned, use `const` instead.

Comment (jmdobry) [reply cli.js:112]: Does making a GET request to this URL, passing the client ID and client Secret actually work?

**Review by JustinBeckwith — APPROVED** [review]

File: cli.js [hunk 19]
```diff
@@ -0,0 +1,263 @@
...
+
+
+
+/**
+ * Creates a csv file of issue data
+ *
+ * @param {array} issues - The issues to group.
+ */
+function makeCSV(issues, file) {
+  const fields = ['repoName', 'issueNumber', 'title', 'body', 'labels'];
```
Comment (jmdobry) [inline cli.js:141]: This CSV file feeds into the model right? Do you actually want `repoName` and `issueNumber` in the dataset, could they affect the model in some way?

File: cli.js [hunk 20]
```diff
@@ -0,0 +1,263 @@
...
+
+  const json2csvParser = new Json2csvParser({ fields, unwind: 'labels' });
+  const csv = json2csvParser.parse(issues);
+
+  fs.writeFileSync(file, csv);
+}
+
+
+//
```
Comment (jmdobry) [inline cli.js:150]: Probably want to remove all commented-out code from this PR

File: defaultsettings.json [hunk 21]
```diff
@@ -0,0 +1,6 @@
+{
+  "githubClientID": ["YOUR GITHUB CLIENT ID HERE"],
```
Comment (jmdobry) [inline defaultsettings.json:2]: I don't think you want the values for these variables to all be wrapped in arrays

File: okiedokie2.csv [hunk 22]
```diff
@@ -0,0 +1,285 @@
+"repoName","issueNumber","title","body","labels"
```
Comment (jmdobry) [inline okiedokie2.csv:1]: I don't think you wanted to commit this file?

File: okiedokie.csv [hunk 23]
```diff
@@ -0,0 +1 @@
+"repoName","issueNumber","title","body","labels"
```
Comment (jmdobry) [inline okiedokie.csv:1]: I don't think you wanted to commit this file?

File: testIssues.json [hunk 24]
```diff
@@ -0,0 +1,150 @@
+{
```
Comment (jmdobry) [inline testIssues.json:1]: I don't think you wanted to commit this file? Or do you actually have some tests for the CLI that use this test data?

**Pull request merged**